### PR TITLE
8325082: Rename headers named 'heapRegion*' of G1

### PR DESCRIPTION
**Pull request by LizBing**
---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8325082](https://bugs.openjdk.org/browse/JDK-8325082): Rename headers named 'heapRegion*' of G1 (**Enhancement** - P4)

### Reviewers
 * [Stefan Karlsson](https://openjdk.org/census#stefank) (@stefank - **Reviewer**) ⚠️ Review applies to [7d105a58](https://git.openjdk.org/jdk/pull/17666/files/7d105a584d378bf58a38330ac2d51d2df75c9ff2)
 * [Thomas Schatzl](https://openjdk.org/census#tschatzl) (@tschatzl - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17666/head:pull/17666` \
`$ git checkout pull/17666`

Update a local copy of the PR: \
`$ git checkout pull/17666` \
`$ git pull https://git.openjdk.org/jdk.git pull/17666/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17666`

View PR using the GUI difftool: \
`$ git pr show -t 17666`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17666.diff">https://git.openjdk.org/jdk/pull/17666.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17666#issuecomment-1920466188)

### CODE REVIEW

File: src/hotspot/cpu/aarch64/gc/g1/g1BarrierSetAssembler_aarch64.cpp [hunk 1]
```diff
@@ -29,7 +29,7 @@
 #include "gc/g1/g1BarrierSetRuntime.hpp"
 #include "gc/g1/g1CardTable.hpp"
 #include "gc/g1/g1ThreadLocalData.hpp"
-#include "gc/g1/heapRegion.hpp"
+#include "gc/g1/g1HeapRegion.hpp"
```
Comment (dholmes-ora) [inline src/hotspot/cpu/aarch64/gc/g1/g1BarrierSetAssembler_aarch64.cpp:32]: Drive by comment, but if you make this change you also have to move it to preserve the include order.

File: src/hotspot/share/gc/g1/g1HeapRegion.hpp [hunk 2]
```diff
@@ -22,14 +22,14 @@
...
 
-#ifndef SHARE_GC_G1_HEAPREGION_HPP
-#define SHARE_GC_G1_HEAPREGION_HPP
+#ifndef SHARE_GC_G1_G1HEAPREGION_HPP
+#define SHARE_GC_G1_G1HEAPREGION_HPP
 
 #include "gc/g1/g1BlockOffsetTable.hpp"
+#include "gc/g1/g1HeapRegionTracer.hpp"
+#include "gc/g1/g1HeapRegionType.hpp"
 #include "gc/g1/g1HeapRegionTraceType.hpp"
```
Comment (stefank) [inline src/hotspot/share/gc/g1/g1HeapRegion.hpp:31]: Sort order needs to be fixed here.

File: src/hotspot/share/gc/g1/g1RemSet.cpp [hunk 3]
```diff
@@ -33,16 +33,16 @@
 #include "gc/g1/g1ConcurrentRefine.hpp"
 #include "gc/g1/g1DirtyCardQueue.hpp"
 #include "gc/g1/g1FromCardCache.hpp"
+#include "gc/g1/g1HeapRegion.inline.hpp"
+#include "gc/g1/g1HeapRegionManager.inline.hpp"
+#include "gc/g1/g1HeapRegionRemSet.inline.hpp"
```
Comment (stefank) [inline src/hotspot/share/gc/g1/g1RemSet.cpp:38]: Sort order. Should go after g1GC.

**Review by stefank — APPROVED** [review]
Looks good to me. There's just a few sort-order issues that needs to be fixed. Please, wait for one of the G1 devs to also review the patch.

**Comment by albertnetymk** [comment]
`src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/gc/g1/HeapRegion.java` and related files should be updated as well.

**Comment by LizBing** [comment]
> `src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/gc/g1/HeapRegion.java` and related files should be updated as well.

I think we should let the cpp part go first. Could you please tell me the relationship between 'jdk.hotspot.agent' and the source files of hotspot?

**Comment by albertnetymk** [comment]
>  Could you please tell me the relationship between 'jdk.hotspot.agent' and the source files of hotspot?

I overlooked that this PR does only file renaming -- class names stay the same. The change inside `jdk.hotspot.agent` should be reverted for this PR.  (They should be done along with the class renaming, e.g. `HeapRegion` -> `G1HeapRegion`.) Sorry for the confusion.

**Review by tschatzl — APPROVED** [review]

**Comment by LizBing** [comment]
thanks for the review.
/integrate

**Pull request closed without merging**